lib/network/generate_key.go

```diff
@@ -0,0 +1,156 @@
+package sebaknetwork
+
+import (
+	"crypto/ecdsa"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"math/big"
+	"os"
+	"path/filepath"
+	"time"
+)
+
+var (
+	host     = "localhost"
+	validFor = time.Duration(1000000)
+	rsaBits  = 2048
+)
+
+func publicKey(priv interface{}) interface{} {
+	switch k := priv.(type) {
+	case *rsa.PrivateKey:
+		return &k.PublicKey
+	case *ecdsa.PrivateKey:
+		return &k.PublicKey
+	default:
+		return nil
+	}
+}
+
+func pemBlockForKey(priv interface{}) *pem.Block {
+	switch k := priv.(type) {
+	case *rsa.PrivateKey:
+		return &pem.Block{Type: "RSA PRIVATE KEY", Bytes: x509.MarshalPKCS1PrivateKey(k)}
+	case *ecdsa.PrivateKey:
+		b, err := x509.MarshalECPrivateKey(k)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Unable to marshal ECDSA private key: %v", err)
+			os.Exit(2)
+		}
+		return &pem.Block{Type: "EC PRIVATE KEY", Bytes: b}
+	default:
+		return nil
+	}
+}
+
+type KeyGenerator struct {
+	certPath string
+	keyPath  string
+}
+
+var (
+	tlsDirPath  = "tls_tmp"
+	tlsPrefix   = "sebak"
+	certPostfix = ".cert"
+	keyPostfix  = ".key"
+)
+
+func remove(filePath string) {
+	if _, err := os.Stat(filePath); os.IsNotExist(err) {
```

Comment:
Is it possible to put the `os.IsNotExist` check after the call to `Remove`, or does `Remove` not return that error ?

Reply:
Avoid a redundant system call, makes code smaller and simpler to follow.

Reply:
@Geod24 It is natural to check the existence before removing file. To reduce the system calls seems to be good, but if I have a right to select, I will select the existing code because in this case keeping the legibility(or understanding) of logics will be better.

Reply:
I'm fine with leaving it as is then